packages/paste-core/components/form/src/shared/FormControlWrapper.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {FieldWrapper} from './FieldWrapper';
+import {Prefix} from './Prefix';
+import {Suffix} from './Suffix';
+import {safelySpreadFormControlProps} from './Utils';
+
+interface FormControlWrapperProps {
+  children: NonNullable<React.ReactNode>;
+  disabled?: boolean;
+  hasError?: boolean;
+  insertAfter?: React.ReactNode;
+  insertBefore?: React.ReactNode;
+}
+
+const FormControlWrapper: React.FC<FormControlWrapperProps> = ({
```

Comment:
Can you use this in Input, Select and Textarea as it’s designed to be a shared things for all?

Reply:
Yes. I did some testing with Input, and there's just some slight restructuring that will need to happen with that as well as Select and Textarea. I think we could get rid of FieldWrapper after that refactor. Adding a Type prop for use with Input later.